recipes/iir1/all/patches/0001-Different-export-headers-for-shared-and-static-lib.patch

```diff
@@ -0,0 +1,32 @@
+From 0ddff567ef8f513462f0846bcdd481774a01da95 Mon Sep 17 00:00:00 2001
```

Comment:
This patch is upstreamable, if you add the following to the cmake script as well:
```cmake
target_compile_definitions(iir PUBLIC iir_SHARED)
```

Reply:
I don't this this patch is applied in the current pr. Is it necessary?

Reply:
I took another look at the logs, and the cmake scripts installs both shared and static libraries

Reply:
When addressing https://github.com/conan-io/conan-center-index/pull/8171#pullrequestreview-821867768, I think this comment might become relevant:
you'll need to add `iir_SHARED` to `package_info`.

Reply:
Is there a core issue with both the shared and static lib being build at the same time? If so, which approach is preferred: deleting one of the libs, or adding a patch introducing and using an option like `BUILD_SHARED_LIBS` to only build one lib at a time?

Reply:
Personally, I always choose the solution of least effort or the one that is most likely to be accepted by upstream.
Using `BUILD_SHARED_LIBS` is idiomatic cmake, but will break the current behavior.

I would choose `BUILD_SHARED_LIBS`, and submit the patch upstream.
I'm pushing for pushing these upstream so we won't have to maintain intrusive patches.

Reply:
I do agree with you that `BUILD_SHARED_LIBS` is the way to go. Ill try to get these changes to be accepted upstream, and create an update for this recipe once that has happened.
But since there is no timeline or this, or if it will even be accepted ill I will add an update deleting the unused lib for now, and create an updated once its been upstreamed.

Reply:
To get things moving, I would create a small patch for upstream, that can be re-used here.
It's completely fine to have the iir1 recipe merged here before upstream accepts your pr.